hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -174,6 +199,12 @@ return_type GenericSystem::read()
 {
   // do loopback
   joint_states_ = joint_commands_;
+  for (const auto & mimic_joint : mimic_joints_) {
+    for (auto i = 0u; i < standard_interfaces_.size(); ++i) {
```

Comment:
```suggestion
    for (auto i = 0u; i < joint_states_.size(); ++i) {
```
even though the code promises this to be that size, it is much more bulletproof to directly iterate over the size of the vector

Reply:
The we could also mimic non-standard interfaces :D Fun :D